pkg/roachprod/failureinjection/failures/node_kill.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package failures
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"golang.org/x/sys/unix"
+)
+
+type NodeKillArgs struct {
+	Nodes install.Nodes
+	// GracefulShutdown will allow the cockroach process to drain before exiting.
+	GracefulShutdown bool
+}
+
+const NodeKillFailureName = "node-kill"
+
+func registerNodeKillFailure(r *FailureRegistry) {
+	r.add(NodeKillFailureName, NodeKillArgs{}, MakeNodeKillFailure)
+}
+
+func MakeNodeKillFailure(clusterName string, l *logger.Logger, secure bool) (FailureMode, error) {
+	c, err := roachprod.GetClusterFromCache(l, clusterName, install.SecureOption(secure))
+	if err != nil {
+		return nil, err
+	}
+
+	return &NodeKillFailure{
+		GenericFailure: GenericFailure{
+			c:        c,
+			runTitle: NodeKillFailureName,
+		},
+	}, nil
+}
+
+type NodeKillFailure struct {
```

Comment:
Should this be renamed to ProcessKillFailure to differentiate from an upcoming VMKillFailure? (The latter also has two modes of "shutdown", at least in GCE.)